src/ast/irbuilderbpf.cpp

```diff
@@ -591,10 +591,9 @@ Value *IRBuilderBPF::CreateUSDTReadArgument(Value *ctx,
     if (argument->valid & BCC_USDT_ARGUMENT_DEREF_OFFSET) {
       Value *ptr = CreateAdd(CreateLoad(getInt64Ty(), reg),
                              getInt64(argument->deref_offset));
-      // Zero out `dst` here in case we read less than 64 bits
-      CreateStore(getInt64(0), dst);
       CreateProbeRead(ctx, dst, abs_size, ptr, as, loc);
-      result = CreateLoad(dst);
+      result = CreateLoad(GetType(CreateInt(abs_size * 8)), dst);
```

Comment:
do we have any codegen tests for this? Is the IR still valid with this change? ( can validate this with llvm-as, see the codegen validator test)

Reply:
ok. will check